gobblin-modules/gobblin-azkaban/src/main/java/org/apache/gobblin/service/modules/orchestration/AzkabanSpecProducer.java

```diff
@@ -84,6 +83,18 @@ public void close() throws IOException {
   public Future<?> addSpec(Spec addedSpec) {
     // If project already exists, execute it
     try {
+      // If encountered Azkaban authentication issue on start up, attempt authentication again.
+      // If still fails, only fail this specific flow
+      if (_sessionId == null) {
+        try {
+          attemptAzkabanAuthentication();
+        }
+        catch (RuntimeException e) {
```

Comment:
Usually we don't need to catch and rethrow the same type of exception, we can try leaving this code as-is for now and letting the Azkaban client handle this.